supplemental_notebooks/2022-09-20_muddiest_points.Rmd

```diff
@@ -0,0 +1,163 @@
+
+## Setup 
+
+```{r}
+library(magrittr)
+library(ggplot2)
+library(SingleCellExperiment)
+```
+## Understanding UMIs
+
+When preparing a single-cell library for sequencing using the tag based methods discussed in class, individual cells are encapsulated in droplets. 
+Inside that droplet is a bead containing thousands of barcoded oligonucleotides that consist of a common cell barcode and a unique molecular identifier (UMI).
+As the cell is lysed, RNA transcripts will be tagged with a single oligonucleotide so that RNA from a given cell will all obtain the same cell barcode and their own UMI. 
+No two RNA transcripts from a given cell will contain the same UMI so that all of the transcripts are now unique and the UMI can be used to identify all original transcripts that were present in the cell at the time that RNA was isolated for sequencing. 
+
+In order to sequence the RNA that you have isolated, it will need to be amplified through PCR amplification. 
+All of the RNA in each of the cells will be pooled together and amplified using a common primer so that you now have a pool of RNA where there are duplicates of the original RNA, and therefore multiple copies of the same UMI. 
+When quantifying gene expression in the dataset, all transcripts that have the same UMI are actually an artifact of PCR amplification, because during initial capture, no two RNA molecules should have the same UMI. This means that any transcripts with the same UMI should be counted only as a single transcript because they are merely just PCR copies of the same original transcript and do not accurately reflect gene expression. 
+To quantify gene expression of a particular gene in a given cell we count up the total number transcripts that map to that gene and contain unique UMIs, discarding any duplicate UMIs as they are artifacts of PCR amplification. 
+The final value that you see in the counts matrix will correspond to that total number of unique UMIs, after duplicates have removed. 
+
+Let's read in the filtered data saved after performing filtering of the the bladder sample in `02-filtering_scRNA.Rmd` and take a closer look at the counts data:
+
+```{r}
+# file paths to filtered sce 
+data_dir <- file.path("data", "tabula-muris")
+filtered_sce_file <- file.path(data_dir, "filtered", "filtered_sce.rds")
+
+# read in filtered sce 
+filtered_bladder_sce <- readr::read_rds(filtered_sce_file)
+```
+
+
+```{r}
+counts(filtered_bladder_sce)[1:5, 1:5]
+```
+
+Here we are looking at the first 5 columns and the first 5 rows of the counts matrix. 
+You can see that the names of the columns contain nucleotide strings, these are the cell barcodes. 
+This means that for any gene that has expression > 0 in that column, they had transcripts that were tagged with that same cell barcode and were from the same original cell. 
+
+We can also look at individual genes and identify how many copies of each of the genes were originally present in the cell, PRIOR to any PCR amplification. 
```

Comment:
since PCR is in caps it looked a little weird, so changed to italics - 

```suggestion
We can also look at individual genes and identify how many copies of each of the genes were originally present in the cell, _prior_ to any PCR amplification. 
```